test/e2e/traffic_test.go

```diff
@@ -69,12 +69,15 @@ func TestTraffic(t *testing.T) {
 		})
 
 		ctx.NewSubTest("requests to c should fail").Run(func(ctx framework.TestContext) {
-			a[0].CallOrFail(t, echo.CallOptions{
+			res, err := a[0].Call(echo.CallOptions{
 				Address: fmt.Sprintf("c.%s.svc.cluster.local", appNs.Name()),
 				Port:    ports.HTTP,
-				Check:   check.Status(503),
+				Check:   check.Error(),
 				Timeout: 1 * time.Second,
 			})
+			if err == nil || res.Responses.Len() != 0 {
+				t.Fatalf("the request did not fail and got the following response: %v", res)
+			}
```

Comment:
Again, n00b lenses:

- Wouldn't `Check:   check.Error()` above imply that  if we reach this line the `err` is always `nil`?  
- What if we get an empty response? Is it possible at all? Should we `Fatalf`  in such a case too?

Reply:
Thanks! I forgot about that check. I removed it.